packages/core/src/app/payment/Payment.tsx

```diff
@@ -97,12 +97,20 @@ class Payment extends Component<PaymentProps & WithCheckoutPaymentProps & WithLa
             onReady = noop,
             onUnhandledError = noop,
             usableStoreCredit,
+            defaultMethod,
+            emitAnalyticsEvent,
         } = this.props;
 
+        const { selectedMethod = defaultMethod } = this.state;
```

Comment:
Is there a reason to store this in state? When we call `emitAnalyticsEvent`, can't we just call defaultMethod from `this.props`?

Reply:
this is just to say that the `selectedMethod` defaults to `defaultMethod` if undefined and I didn't want to assume that selectedMethod is always undefined at this point

Reply:
ah gotcha, misread this